plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -33,23 +43,47 @@ func (is *InternetSpeed) SampleConfig() string {
 
 const measurement = "internet_speed"
 
+var serverCache *speedtest.Server
+
 func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
-	user, err := speedtest.FetchUserInfo()
-	if err != nil {
-		return fmt.Errorf("fetching user info failed: %v", err)
-	}
-	serverList, err := speedtest.FetchServerList(user)
-	if err != nil {
-		return fmt.Errorf("fetching server list failed: %v", err)
+
+	// Sleep for offset duration
+	if is.Offset != "" {
+		d, err := time.ParseDuration(is.Offset)
+		if err != nil {
+			return fmt.Errorf("parsing offset failed: %v", err)
+		}
+		time.Sleep(d)
 	}
 
-	if len(serverList.Servers) < 1 {
-		return fmt.Errorf("no servers found")
+	// Get closest server
+	var s *speedtest.Server
+	if serverCache != nil {
+		is.Log.Debug("Using cached server")
+		s = serverCache
+	} else {
+		user, err := speedtest.FetchUserInfo()
+		if err != nil {
+			return fmt.Errorf("fetching user info failed: %v", err)
+		}
+		serverList, err := speedtest.FetchServerList(user)
+		if err != nil {
+			return fmt.Errorf("fetching server list failed: %v", err)
+		}
+		if len(serverList.Servers) < 1 {
+			return fmt.Errorf("no servers found")
+		}
+		s = serverList.Servers[0]
+		is.Log.Debug("Found server: ", s.String())
+		if is.Cache {
+			serverCache = s
+		}
 	}
-	s := serverList.Servers[0]
+
+	// Run the tests
```

Comment:
Unnecessary comment. That's exactly what the debug message one line down says...